client-src/elements/chromedash-form-field.js

```diff
@@ -117,52 +129,100 @@ export class ChromedashFormField extends LitElement {
     };
     this.dispatchEvent(new CustomEvent('form-field-update', eventOptions));
 
-    // Run semantic check on this field.  Must be after above dispatch.
-    this.doSemanticCheck();
-    // Also doSemanticCheck on known dependent form fields.
+    // Run semantic checks on entire page.  Must be after above dispatch.
     const app = document.querySelector('chromedash-app');
-    const dependents = ALL_FIELDS[this.name].dependents;
-    if (dependents && app?.pageComponent) {
-      dependents.forEach((dependent) => {
-        const dependentField = app.pageComponent.allFormFieldComponents[dependent];
-        if (dependentField) {
-          dependentField.doSemanticCheck();
-        }
-      });
+    if (app?.pageComponent) {
+      app.pageComponent.allFormFieldComponentsList.forEach((formFieldComponent) =>
+        formFieldComponent.doSemanticCheck());
+    } else {
+      // Do the semantic check for unit testing.  Only works for isolated field.
+      this.doSemanticCheck();
```

Comment:
Can we run this on initial page load as well? When I tested this and had separate bad inputs already submitted for my feature, I would see warnings start to populate on fields I wasn't editing. Just running this and adding the warnings on page load should fix that unusual behavior.

Reply:
There is a call of this.doSemanticCheck() in firstUpdated.  This should result in doing the checks when the page is loaded.  But your comment reminds me that I should add a test for that.  (It worked in my manual tests, but it needs to be in the tests.)

Reply:
I am still seeing the issue, but specifically when editing an OT stage or shipping stage in `chromedash-edit-stage-page`. I think this is because the shipping/OT milestones for the stage NOT being edited are not a part of the form.

See the below video where I begin editing a shipping stage when the the OT and shipping milestones have already been submitted with invalid values (OT desktop milestone of 123 and shipping desktop milestone of 120). The warning will pop up only after I've edited any other field on the page.

I think this can be easily fixed by doing a run of the semantic checks when the component is loaded.

https://github.com/GoogleChrome/chromium-dashboard/assets/56164590/decc66d3-0c02-4fb3-ba12-211ad11f930b

Reply:
Great fix, though it seems there ought to be a cleaner way to deal with these kinds of things.